### DISCUSSION TITLE
Welcome to sail Discussions!

### DISCUSSION THREAD

**Discussion by PeterSewell** [rems-project/sail#348] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by rsnikhil** [comment]
Hello, I'm Rishiyur Nikhil. Co-founder and CTO of Bluespec, Inc. Was chair of RISC-V group that selected Sail for the official RISC-V formal spec (some years ago).

Now volunteering with "RVI Golden Model" group, chaired by Bill McSpadden, to improve/complete/update the Sail RISC-V spec.

Am particularly interested in Sail's type-system, and comparing it with Bluespec, which has a Haskell-like type-system.  Bluespec also has numeric Kinds to support fixed-size entities like vectors, bit-vectors etc.  Like Sail, these can be polymorphic, but Bluespec uses Haskell-style typeclasses for arithmetic constraints, etc.).  Bluespec also uses Yices (or STP) to solve constraints.
